lamindb_setup/_schema_metadata.py

```diff
@@ -0,0 +1,472 @@
+import hashlib
+import importlib
+import json
+from dataclasses import dataclass
+from typing import Dict
+from uuid import UUID
+
+import sqlparse
+from django.contrib.postgres.expressions import ArraySubquery
+from django.db.models import (
+    Field,
+    ForeignObjectRel,
+    ManyToManyField,
+    ManyToManyRel,
+    OuterRef,
+    QuerySet,
+    Subquery,
+)
+from django.db.models.functions import JSONObject
+from lnschema_core.models import LinkORM, Registry
+from sqlparse.sql import Identifier, IdentifierList
+from sqlparse.tokens import DML, Keyword
+from supabase import Client
+
+from lamindb_setup import settings
+from lamindb_setup._init_instance import get_schema_module_name
+from lamindb_setup.core._hub_client import call_with_fallback_auth
+
+
+def synchronize_schema():
+    return call_with_fallback_auth(_synchronize_schema)
+
+
+def _synchronize_schema(client: Client):
+    schema_metadata_dict = schema_metadata.to_json()
+
+    schema_encoded = json.dumps(schema_metadata_dict, sort_keys=True).encode("utf-8")
```

Comment:
Alternatively, consider bcoding the dictionary prior to hashing it, similar to: https://github.com/insitro/redun/blob/367033e87ad17ad8a1bdb09bec472f846ffe4b4e/redun/bcoding.py#L206-L212